src/Strategy/AbstractStrategy.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Http\Discovery\Strategy;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+abstract class AbstractStrategy implements DiscoveryStrategy
+{
+    protected static $classes = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function find($type)
+    {
+        if (isset(static::$classes[$type])) {
+            return static::$classes[$type];
```

Comment:
> If strategies dont do the test, they dont know if the current resource is available and cant proceed to the next one.

Im not sure that is true anymore. Each strategy return all *candidates* they know of. It is the `ClassDiscovery` that makes a decision if we should use a candidate and if we should visit the next strategy.